src/modules/sync/sync/languages.ts

```diff
@@ -0,0 +1,149 @@
+import { LanguageModels, ManagementClient } from "@kontent-ai/management-sdk";
+import { match } from "ts-pattern";
+import { v4 as createUuid } from "uuid";
+import { z } from "zod";
+
+import { logInfo, LogOptions } from "../../../log.js";
+import { omit } from "../../../utils/object.js";
+import { serially } from "../../../utils/requests.js";
+import { DiffModel } from "../types/diffModel.js";
+import { PatchOperation } from "../types/patchOperation.js";
+
+export const syncLanguages = async (
+  client: ManagementClient,
+  operations: DiffModel["languages"],
+  logOptions: LogOptions,
+) => {
+  logInfo(logOptions, "standard", "Syncing Languages");
+
+  logInfo(logOptions, "standard", "Adding Languages");
+
+  await serially(operations.added.filter(op => !op.is_default).map(l => () => addLanguage(client, l)));
+
+  logInfo(logOptions, "standard", "Updating Languages");
+  const transformedOperations = [...operations.updated].map(
+    ([codename, operations]) => [codename, operations.map(transformLanguagePatchOperation)] as const,
+  );
+
+  const sortedOperations = transformedOperations.toSorted(([, operations], [, operations2]) =>
+    operationsToOrdNumb(operations) - operationsToOrdNumb(operations2)
+  );
+
+  await serially(
+    sortedOperations.map(([codename, operations]) => () => modifyLanguage(client, codename, operations)),
+  );
+
+  logInfo(logOptions, "standard", "Deleting languages");
+
+  await serially(
+    [...operations.deleted].map(codename => () => deleteLanguage(client, codename)),
+  );
+};
+
+const addLanguage = (client: ManagementClient, langauge: LanguageModels.IAddLanguageData) =>
+  client
+    .addLanguage()
+    .withData(langauge)
+    .toPromise();
+
+const modifyLanguage = (
+  client: ManagementClient,
+  codename: string,
+  operations: LanguageModels.IModifyLanguageData[],
+) =>
+  client
+    .modifyLanguage()
+    .byLanguageCodename(codename)
+    .withData(operations)
+    .toPromise();
+
+const deleteLanguage = (client: ManagementClient, codename: string) => {
+  const randomUuid = createUuid();
+  return client
+    .modifyLanguage()
+    .byLanguageCodename(codename).withData([
+      /**
+       * languages cannot be deleted, instead they are deactivated
+       * and their name and codename are both populated with first 8 chars of their ID
+       * (name and codename have a limit of 25 characters).
```

Comment:
We will need to update the whole readme after we add all of the entity.. It would be better to adjust it slightly before the release as it might confuse users with the newly added features that are not released yeat